packages/loader/container-loader/src/container.ts

```diff
@@ -1642,6 +1686,19 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         } else if (value === ConnectionState.Disconnected) {
             // Important as we process our own joinSession message through delta request
             this.pendingClientId = undefined;
+            // Only wait for "leave" message if we have some outstanding ops and the client was write client as
+            // server would not accept ops from read client.
+            if (this._deltaManager.clientSequenceNumber - this._deltaManager.clientSequenceNumberObserved > 0
+                && this.client.mode === "write"
+            ) {
+                this.prevClientLeftP = new Deferred();
+                // Max time is 5 min for which we are going to wait for its own "leave" message.
```

Comment:
I personally do not see much value in waiting 5 min - it always same as server timeout (server has 5 min timeout, the difference is only in TCP/IP timeout that adds on top of it for worse possible case).
I'm of an opinion that we want to go with much smaller timeout as we are not making things worse then they are today (data corruption). Going smaller will ensure that we can find reasonable place where we feel Ok

Reply:
Changed it to 90 sec. Should we reduce it more?